src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -755,6 +758,11 @@ private BidderRequest createBidderRequest(BidderPrivacyResult bidderPrivacyResul
 
         final App bidRequestApp = bidRequest.getApp();
         final Site bidRequestSite = bidRequestApp == null ? bidRequest.getSite() : null;
+
+        if (bidRequestApp != null && bidRequest.getSite() != null) {
```

Comment:
No, if site and app present `bidRequestSite` will be null, what means if condition will not work as expected

Reply:
ok, that's why i'm asking for because of little cryptic.

Let's make some refactoring though, like:
```
final App app = bidRequest.getApp();
final Site site = bidRequest.getSite();
if (app != null && site != null) {
  // add warning
}

final Site resolvedSite = app == null ? site : null;
```
So, use `resolvedSite` in code below instead.

Reply:
Nice catch, thx)